idl/github.com/uber/cadence/shared.thrift

```diff
@@ -1484,7 +1484,7 @@ struct BatchCancelRequest{
 
 struct StartBatchJobRequest{
     10: optional string domain
-    20: optional string jobID
+    20: optional SearchAttributes searchAttributes
     30: optional BatchOperationType operationType
     40: optional string sourceQuery
     50: optional string reason
```

Comment:
Overall this input request struct looks odd.. 

what is header, requestID, identity and startToCloseTimeoutSeconds used for? 

I think things like sourceQuery, reason, rps should be clubbed into a single struct that applies to batch in general for clarity.. searchAttributes also falls into this bucket..

Reply:
BatchJob will be a first class member as Activity/Timer/ChildWorkflow/etc in our clientSDK. 
Similar to start a workflow:
Header is for jeager propagation. 
RequestedID is for dedup. 
identity is also used for our client SDK.
startToCloseTimeoutSeconds is timeout for the batchjob. 

How do you call the struct? I don't see this is a convention in our IDL yet. Btw there will be another layer above in our client SDK.

Reply:
I prefer to put all the general BatchOperation options into a separate struct or at least improve the ordering here. The general options and then the batch specific options..